PeakNetwork/Core/Endpoint.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  Endpoint.swift
+//  PeakNetwork-iOS
+//
+//  Created by Sam Oakley on 12/03/2019.
+//  Copyright © 2019 3Squared. All rights reserved.
+//
+
+import Foundation
+
+public typealias URLComponentsCustomisationBlock = (inout URLComponents) -> ()
+
+/// Describes an API endpoint.
+public struct Endpoint {
+    let baseURL: String
+    let path: String
+    let queryItems: [String: String]
```

Comment:
This should be `[String: String?]` because `URLQueryItem` can take nil for its value. This should perhaps be made in to a `typealias`

Reply:
I have replace `let queryItems: [String: String]` with `let queryItems: [URLQueryItem]`.